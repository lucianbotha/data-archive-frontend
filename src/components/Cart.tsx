import {
  faWindowClose,
  faDownload,
  faEraser,
  faUserPlus
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import * as React from "react";
import Modal from "react-responsive-modal";
import { NavLink } from "react-router-dom";
import CartFile from "./cart/CartFile";

interface ICart {
  open: boolean;
  openCart: (open: boolean) => void;
  clearCart: () => void;
  user?: any;
}

class Cart extends React.Component<ICart, any> {
  state: {
    cart: any[];
  } = {
    cart: []
  };
  public render() {
    const { open, openCart, clearCart, user } = this.props;
    const c: any = [
      {
        files: [
          {
            declination: "12",
            filename: "filename-1",
            name: "name-1",
            rightAscension: "66",
            targetName: "NGS-1234"
          },
          {
            declination: "12",
            filename: "filename-12",
            name: "name-12",
            rightAscension: "66",
            targetName: "NGS-1234"
          },
          {
            declination: "12",
            filename: "filename-13",
            name: "name-13",
            rightAscension: "66",
            targetName: "NGS-1234"
          }
        ],
        id: "ID-1"
      },
      {
        files: [
          {
            declination: "14",
            filename: "filename-2",
            name: "name-2",
            rightAscension: "63",
            targetName: "NGS-123"
          },
          {
            declination: "14",
            filename: "filename-21",
            name: "name-21",
            rightAscension: "63",
            targetName: "NGS-123"
          },
          {
            declination: "14",
            filename: "filename-22",
            name: "name-22",
            rightAscension: "63",
            targetName: "NGS-123"
          },
          {
            declination: "14",
            filename: "filename-23",
            name: "name-23",
            rightAscension: "63",
            targetName: "NGS-123"
          }
        ],
        id: "ID-2"
      },
      {
        files: [
          {
            declination: "34",
            filename: "filename-3",
            name: "name-3",
            rightAscension: "33",
            targetName: "NGS-323"
          },
          {
            declination: "34",
            filename: "filename-31",
            name: "name-31",
            rightAscension: "33",
            targetName: "NGS-123"
          },
          {
            declination: "14",
            filename: "filename-32",
            name: "name-32",
            rightAscension: "33",
            targetName: "NGS-133"
          }
        ],
        id: "ID-3"
      },
      {
        files: [
          {
            declination: "144",
            filename: "filename-4",
            name: "name-4",
            rightAscension: "43",
            targetName: "NG4-123"
          },
          {
            declination: "44",
            filename: "filename-41",
            name: "name-41",
            rightAscension: "43",
            targetName: "NGS-123"
          },
          {
            declination: "14",
            filename: "filename-42",
            name: "name-42",
            rightAscension: "4",
            targetName: "NGS-143"
          },
          {
            declination: "14",
            filename: "filename-43",
            name: "name-43",
            rightAscension: "63",
            targetName: "NG4-123"
          }
        ],
        id: "ID-4"
      },
      {
        files: [
          {
            declination: "34",
            filename: "filename-3",
            name: "name-3",
            rightAscension: "33",
            targetName: "NGS-323"
          },
          {
            declination: "34",
            filename: "filename-31",
            name: "name-31",
            rightAscension: "33",
            targetName: "NGS-123"
          },
          {
            declination: "14",
            filename: "filename-32",
            name: "name-32",
            rightAscension: "33",
            targetName: "NGS-133"
          }
        ],
        id: "ID-6"
      }
    ];
    return (
      <Modal open={open} onClose={() => openCart(false)} center={true}>
        <div className={"cart-table section"}>
          <div className={"c-table"}>
            <table
              className={"table is-striped is-info is-fullwidth is-outlined"}
            >
              <thead>
                <tr className="notification cart-modal">
                  <th>Observation</th>
                  <th>Remove</th>
                  <th>Filename</th>
                  <th>Name</th>
                  <th>Right ascension</th>
                  <th>Declination</th>
                  <th>Target name</th>
                </tr>
              </thead>
              <tbody>
                {c.map((obz: any) => {
                  return (
                    <>
                      {obz.files.map((file: any, index: number) => (
                        <CartFile
                          observation={obz}
                          index={index}
                          id={obz.id}
                          key={file.filename}
                          file={file}
                        />
                      ))}
                    </>
                  );
                })}
              </tbody>
            </table>
          </div>
        </div>
        <div className={"section"}>
          <div className={"button-scales "}>
            <div className={"columns"}>
              <div className={"column"}>
                {!user ? (
                  <NavLink to={"login"}>
                    <button
                      className={"button is-primary"}
                      onClick={() => openCart(false)}
                    >
                      <span>
                        Login <FontAwesomeIcon icon={faUserPlus} />
                      </span>
                    </button>
                  </NavLink>
                ) : (
                  <NavLink to={"data-requests"}>
                    <button
                      className={"button is-primary"}
                      onClick={() => openCart(false)}
                    >
                      <span>
                        Request <FontAwesomeIcon icon={faDownload} />
                      </span>
                    </button>
                  </NavLink>
                )}
              </div>
              <div className={"column"}>
                <button
                  className={"button is-danger"}
                  onClick={() => openCart(false)}
                >
                  <span>
                    Close <FontAwesomeIcon icon={faWindowClose} />
                  </span>
                </button>
              </div>
              <div className={"column"}>
                <button
                  className={"button is-warning"}
                  onClick={() => clearCart()}
                >
                  <span>
                    Clear <FontAwesomeIcon icon={faEraser} />
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </Modal>
    );
  }
}

export default Cart;
