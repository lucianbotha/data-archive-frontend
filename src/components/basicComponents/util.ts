const updateState = (state: any, name: string, value: string) => {
  let newState = state;
  if (name === "targetName") {
    newState = {
      ...state,
      target: {
        ...state.target,
        targetName: {
          ...state.target.targetName,
          name: value
        }
      }
    };
  }
  if (name === "ra") {
    newState = {
      ...state,
      target: {
        ...state.target,
        ra: {
          ...state.target.ra,
          value
        }
      }
    };
  }
  if (name === "dec") {
    newState = {
      ...state,
      target: {
        ...state.target,
        dec: {
          ...state.target.dec,
          value
        }
      }
    };
  }
  if (name === "radius") {
    newState = {
      ...state,
      target: {
        ...state.target,
        radius: {
          ...state.target.radius,
          value
        }
      }
    };
  }
  if (name === "radiusUnits") {
    newState = {
      ...state,
      target: {
        ...state.target,
        radiusUnits: {
          ...state.target.radiusUnits,
          value
        }
      }
    };
  }
  if (name === "resolver") {
    newState = {
      ...state,
      target: {
        ...state.target,
        resolver: {
          ...state.target.resolver,
          value
        }
      }
    };
  }
  if (name === "proposalCode") {
    newState = {
      ...state,
      proposal: {
        ...state.proposal,
        proposalCode: {
          ...state.proposal.proposalCode,
          code: value
        }
      }
    };
  }
  if (name === "proposalTitle") {
    newState = {
      ...state,
      proposal: {
        ...state.proposal,
        proposalTitle: {
          ...state.proposal.proposalTitle,
          title: value
        }
      }
    };
  }
  if (name === "pi") {
    newState = {
      ...state,
      proposal: {
        ...state.proposal,
        pi: {
          ...state.proposal.pi,
          name: value
        }
      }
    };
  }
  if (name === "obsDate") {
    newState = {
      ...state,
      proposal: {
        ...state.proposal,
        obsDate: {
          ...state.proposal.obsDate,
          value
        }
      }
    };
  }
  if (name === "telescope") {
    newState = {
      ...state,
      telescope: {
        ...state.telescope,
        selectedTelescope: value
      }
    };
  }
  if (name === "instrument") {
    newState = {
      ...state,
      telescope: {
        ...state.telescope,
        selectedInstrument: value
      }
    };
  }
  if (name === "dataType") {
    newState = {
      ...state,
      data: {
        ...state.data,
        selectedDataType: value
      }
    };
  }
  if (name === "arcs") {
    console.log("arcs", value);
    newState = {
      ...state,
      data: {
        ...state.data,
        arcs: !state.data.arcs
      }
    };
  }
  if (name === "biases") {
    newState = {
      ...state,
      data: {
        ...state.data,
        biases: !state.data.biases
      }
    };
  }
  if (name === "standards") {
    newState = {
      ...state,
      data: {
        ...state.data,
        standards: !state.data.standards
      }
    };
  }
  if (name === "flats") {
    newState = {
      ...state,
      data: {
        ...state.data,
        flats: !state.data.flats
      }
    };
  }
  return newState;
};

const checkDateFormatError = (date: string) => {
  let error = "";
  const dateReg = /^[0-3]\d([./-])[0-1]\d\1\d{4}$/;
  if (date !== "" && !date.match(dateReg)) {
    error = "Date format does not match (date should be like [DD-MM-YYYY])";
  }
  return error;
};

const checkNameFormatError = (name: string) => {
  let error = "";
  const nameReg = /^[a-zA-Z0-9 ]*$/;
  if (name !== "" && !name.match(nameReg)) {
    error = "This should only contains characters contained in a name only";
  }
  return error;
};

const checkValueFormatError = (value: string) => {
  let error = "";
  const dateReg = /^[1-9]\d*(\.\d+)?$/;
  if (value !== "" && !value.match(dateReg)) {
    error = "This should be a number";
  }
  return error;
};

const validateQuery = (state: any) => {
  return {
    ...state,
    proposal: {
      obsDate: {
        ...state.proposal.obsDate,
        error: checkDateFormatError(state.proposal.obsDate.value)
      },
      pi: {
        ...state.proposal.pi,
        error: checkNameFormatError(state.proposal.pi.name)
      },
      proposalCode: {
        ...state.proposal.proposalCode,
        error: checkNameFormatError(state.proposal.proposalCode.code)
      },
      proposalTitle: {
        ...state.proposal.proposalTitle,
        error: checkNameFormatError(state.proposal.proposalTitle.title)
      }
    },
    target: {
      dec: {
        ...state.target.dec,
        error: checkValueFormatError(state.target.dec.value)
      },
      ra: {
        ...state.target.ra,
        error: checkValueFormatError(state.target.ra.value)
      },
      radius: {
        ...state.target.radius,
        error: checkValueFormatError(state.target.radius.value)
      },
      radiusUnits: {
        ...state.target.radiusUnits
      },
      resolver: {
        ...state.target.resolver
      },
      targetName: {
        ...state.target.targetName,
        error: checkNameFormatError(state.target.targetName.name)
      }
    }
  };
};
export { updateState, validateQuery };
