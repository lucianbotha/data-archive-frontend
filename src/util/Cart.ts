/**
 * The cart of requested files.
 */
export class Cart {
  constructor(private files: CartFile[]) {}

  /**
   * Check whether the cart contains the given file.
   *
   * Files are compared by their id. So if any cart file has the same id as the
   * given file, the function returns true, irrespective of any other key-value
   * pairs are equal.
   *
   * Parameters
   * ----------
   * file:
   *     The file whose existence in the cart is checked.
   *
   * Returns
   * -------
   * contains:
   *     Whether the cart contains the given file.
   */
  public contains(file: CartFile) {
    return this.files.some(f => f.id === file.id);
  }

  /**
   * Adds or removes files from the cart.
   *
   * A file in the given array is added if it is not in the cart yet, and it is
   * removed if it is in the cart already. The same logic as for the contains
   * method is used to decide whether a file is in the cart.
   *
   * @param files
   */
  public addOrRemove(files: CartFile[]) {
    files.forEach(file => {
      const cartFilesFilesWithoutFile = this.files.filter(
        f => f.id !== file.id
      );
      if (cartFilesFilesWithoutFile.length !== this.files.length) {
        this.files = [...cartFilesFilesWithoutFile];
      } else {
        this.files = [...this.files, file];
      }
    });
  }

  /**
   * Returns the cart files grouped by observation.
   *
   * The files are returned as a map whose keys are the observation ids. Each
   * value is the set of files for the observation which have the corresponding
   * key as their id.
   *
   * Files which are not linked to an observation are collected in a set with
   * the empty string as its key.
   *
   * For example, consider a cart containing the files
   *
   * A_Obs1, B, C_Obs1, D_Obs2, E
   *
   * where A_Obs1 and C_Obs1 are linked to observation with id Obs1, D_Obs2 is
   * linked to an observation with id Obs2 and B and E are not linked to any
   * observation. Then this method returns a map with the following content.
   *
   * {
   *   '': new Set(B, E),
   *   'Obs1': new Set(A_Obs1, C_Obs1),
   *   'Obs2': new Set(D_Obs2)
   * }
   */
  public groupByObservation() {
    const groups = new Map<string, Set<CartFile>>();
    this.files.forEach(file => {
      const key = (file.observation && file.observation.id) || "";
      if (!groups.has(key)) {
        groups.set(key, new Set<CartFile>());
      }
      (groups.get(key) as Set<CartFile>).add(file);
    });

    return groups;
  }
}

/**
 * Interface for a cart file.
 *
 * Properties
 * ----------
 * id:
 *     Unique file id.
 * name:
 *     File name.
 * observation:
 *     Observation to which the file is linked.
 */
export interface CartFile {
  id: string;
  name: string;
  observation?: Observation | null;
}

/**
 * Interface for an observation to which a cart file is linked.
 *
 * Properties
 * ----------
 * id:
 *     Unique observation id.
 * name:
 *     Observation name.
 */
interface Observation {
  id: string;
  name: string;
}
