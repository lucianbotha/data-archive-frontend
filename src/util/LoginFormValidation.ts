/**
 * An interface for the registration inputs
 *
 * Properties:
 * -----------
 * password:
 *     The password, which must have at least 7 characters.
 * username:
 *     The username, which must not contain upper case letters.
 */
export interface ILoginForm {
  password: string;
  username: string;
}

/**
 * Login form validation for the user input.
 *
 * It accepts an object argument as defined by the ILoginForm interface
 *
 * @param loginInput
 *
 * An example of the submitted user input may look like:
 *
 * loginInput = {
 *
 *    password: "SecureandMatchingPassword",
 *
 *    username: "validusername"
 * }
 *
 * It returns an erray of string items if there is any error message or
 * an empty array if there are no items.
 */

export const validateLoginField = (loginInput: ILoginForm) => {
  // An object to Store errors for all fields.
  const errors = {
    password: "",
    username: ""
  };

  // Check if the submitted username is not empty.
  if (!loginInput.username) {
    errors.username = `Username cannot be empty.`;
  }

  // Check if the submitted username contains upper case characters.
  if (loginInput.username !== loginInput.username.toLowerCase()) {
    errors.username = "Username must be in lowercase";
  }

  // Check if the password is secure enough.
  if (loginInput.password.length <= 6) {
    errors.password = "Password should be at least 7 characters long";
  }

  // return an array consisting of error message if any or empty.
  return errors;
};
