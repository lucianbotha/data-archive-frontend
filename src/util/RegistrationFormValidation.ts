import { validate } from "isemail";
/**
 * An interface for the registration inputs
 *
 * Properties:
 * -----------
 * affiliation:
 *     The affiliation of the user, such as a university or an institute.
 * confirmPassword:
 *      A matching password that must match with the original password.
 * email:
 *     Email address. This will be stored in lower case.
 * familyName:
 *     The family name (surname).
 * givenName:
 *     The given name (first name).
 * password:
 *     The password, which must have at least 7 characters.
 * username:
 *     The username, which must not contain upper case letters.
 */
export interface IRegistrationForm {
  affiliation: string;
  confirmPassword: string;
  email: string;
  givenName: string;
  familyName: string;
  password: string;
  username: string;
}

/**
 * Registration form validation for the user input.
 *
 * It accepts an object argument as defined by the IRegistrationForm interface
 *
 * @param registrationInput
 *
 * An example of the submitted user input may look like:
 *
 * registrationInput = {
 *    affiliation: "University of Cape Town",
 *
 *    confirmPassword: "SecureandMatchingPassword",
 *
 *    emailAddress: "valid@gmail.com",
 *
 *    familyName: "Smith",
 *
 *    givenName: "John",
 *
 *    password: "SecureandMatchingPassword",
 *
 *    username: "validusername"
 * }
 *
 * It returns an erray of string items if the is any error message or
 * an empty array if there are no items.
 */

export const validateRegistrationField = (
  registrationInput: IRegistrationForm
) => {
  // An object to Store errors for all fields.
  const errors: string[] = [];

  // Check if the submitted given name is not empty.
  if (!registrationInput.givenName) {
    errors.push("Given name can't be empty");
  }

  // Check if the submitted family name is not empty.
  if (!registrationInput.familyName) {
    errors.push("Family name can't be empty");
  }

  // Check if the submitted username is not empty.
  if (!registrationInput.username) {
    errors.push(`Username cannot be empty.`);
  }

  // Check if the submitted username contains upper case characters.
  if (registrationInput.username !== registrationInput.username.toLowerCase()) {
    errors.push("Username must be in lowercase");
  }

  // Check if the submitted email address is valid.
  if (!validate(registrationInput.email, { minDomainAtoms: 2 })) {
    errors.push("Email address is invalid");
  }

  // Check if the password is secure enough.
  if (registrationInput.password.length <= 6) {
    errors.push("Password should be at least 7 characters long");
  }

  // Check if passwords match
  if (registrationInput.password !== registrationInput.confirmPassword) {
    errors.push("Passwords do not match");
  }

  // return an array consisting of error message if any or empty.
  return errors;
};
