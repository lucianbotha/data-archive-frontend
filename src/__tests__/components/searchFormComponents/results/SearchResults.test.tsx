import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import SearchResultsTable from "../../../../components/searchFormComponents/results/SearchResultsTable";
import { MockedProvider } from "react-apollo/test-utils";
import { Cart } from "../../../../util/Cart";

const updateCart = jest.fn();

describe("Search results", () => {
  it("should render", () => {
    expect(
      mount(
        <MockedProvider>
          <SearchResultsTable
            searchResults={[]}
            cart={new Cart([])}
            updateCart={updateCart}
          />
        </MockedProvider>
      )
    ).toBeDefined();
  });

  it("should create a table even if there is no search results with only the header", () => {
    const wrapper = mount(
      <MockedProvider>
        <SearchResultsTable
          searchResults={[]}
          cart={[]}
          updateCart={updateCart}
        />
      </MockedProvider>
    );
    expect(wrapper.find("table.table")).toBeTruthy();
    expect(wrapper.find("td").length).toEqual(0);
  });
  it("should create a table with a some row if there is one observation and a file", () => {
    const observation = {
      files: [
        {
          category: "cat-5",
          dataType: "type-1",
          declination: "20",
          filename: "filename-1",
          instrument: "RSS",
          isReduced: true,
          name: "name-1",
          observationNight: "2010-01-01",
          proposalCode: "code-1",
          rightAscension: "20",
          targetName: "target 001",
          telescope: "SALT",
          url: "./image0.jpg"
        }
      ],
      id: "obsID-1",
      name: "obs1",
      proposal: "Code-1",
      startTime: "2018-02-02 17:55:23",
      telescope: "SALT"
    };
    const wrapper = mount(
      <MockedProvider>
        <SearchResultsTable
          searchResults={[observation]}
          cart={new Cart([])}
          updateCart={updateCart}
        />
      </MockedProvider>
    );
    // row for the observation data
    // row for the file head
    // row for the file data
    expect(wrapper.find("tr").length).toEqual(3);
  });
  it("should render data correctly", () => {
    const observation = [
      {
        files: [
          {
            category: "cat-5",
            dataType: "type-1",
            declination: "20",
            filename: "filename-1",
            instrument: "RSS",
            isReduced: true,
            name: "name-1",
            observationNight: "2010-01-01",
            proposalCode: "code-1",
            rightAscension: "20",
            targetName: "target 001",
            telescope: "SALT",
            url: "./image0.jpg"
          }
        ],
        id: "obsID-1",
        name: "obs1",
        proposal: "Code-1",
        startTime: "2018-02-02 17:55:23",
        telescope: "SALT"
      },
      {
        files: [
          {
            category: "cat-1",
            dataType: "type-2",
            declination: "20",
            filename: "filename-2",
            instrument: "RSS",
            isReduced: true,
            name: "name-2",
            observationNight: "2010-01-01",
            proposalCode: "code-2",
            rightAscension: "20",
            targetName: "target 002",
            telescope: "SALT",
            url: "./image4.jpg"
          }
        ],
        id: "obsID-2",
        name: "obs1",
        proposal: "Code-2",
        startTime: "2018-02-02 17:55:23",
        telescope: "SALT"
      }
    ];
    const wrapper = mount(
      <MockedProvider>
        <SearchResultsTable
          searchResults={observation}
          cart={new Cart([])}
          updateCart={updateCart}
        />
      </MockedProvider>
    );
    expect(toJson(wrapper.find("SearchResultsTable"))).toMatchSnapshot();
  });
});
