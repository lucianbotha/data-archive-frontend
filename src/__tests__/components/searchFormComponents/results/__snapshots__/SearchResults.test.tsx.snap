// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search results should render data correctly 1`] = `
.c0.c0 {
  width: 18px;
  height: 18px;
}

<SearchResults
  cart={Array []}
  searchResults={
    Array [
      Object {
        "files": Array [
          Object {
            "category": "cat-5",
            "dataType": "type-1",
            "declination": "20",
            "filename": "filename-1",
            "instrument": "RSS",
            "isReduced": true,
            "name": "name-1",
            "observationNight": "2010-01-01",
            "proposalCode": "code-1",
            "rightAscension": "20",
            "targetName": "target 001",
            "telescope": "SALT",
            "url": "./image0.jpg",
          },
        ],
        "id": "obsID-1",
        "name": "obs1",
        "proposal": "Code-1",
        "startTime": "2018-02-02 17:55:23",
        "telescope": "SALT",
      },
      Object {
        "files": Array [
          Object {
            "category": "cat-1",
            "dataType": "type-2",
            "declination": "20",
            "filename": "filename-2",
            "instrument": "RSS",
            "isReduced": true,
            "name": "name-2",
            "observationNight": "2010-01-01",
            "proposalCode": "code-2",
            "rightAscension": "20",
            "targetName": "target 002",
            "telescope": "SALT",
            "url": "./image4.jpg",
          },
        ],
        "id": "obsID-2",
        "name": "obs1",
        "proposal": "Code-2",
        "startTime": "2018-02-02 17:55:23",
        "telescope": "SALT",
      },
    ]
  }
  updateCart={[MockFunction]}
>
  <_default
    closeModal={[Function]}
    image={
      Object {
        "alt": "Some text to show",
        "url": "",
      }
    }
    open={false}
  >
    <Modal
      animationDuration={500}
      blockScroll={true}
      center={true}
      classNames={Object {}}
      classes={
        Object {
          "closeButton": "styles_closeButton__20ID4",
          "closeIcon": "styles_closeIcon__1QwbI",
          "modal": "styles_modal__gNwvD",
          "overlay": "styles_overlay__CLSq-",
          "overlayCenter": "styles_overlayCenter__YHoO7",
          "transitionEnter": "styles_transitionEnter__3j_-a",
          "transitionEnterActive": "styles_transitionEnterActive___eQs7",
          "transitionExit": "styles_transitionExit__1KmEf",
          "transitionExitActive": "styles_transitionExitActive__1nQXw",
        }
      }
      closeIconId={null}
      closeIconSize={28}
      closeIconSvgPath={
        <path
          d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
        />
      }
      closeOnEsc={true}
      closeOnOverlayClick={true}
      focusTrapOptions={Object {}}
      focusTrapped={false}
      modalId={null}
      onClose={[Function]}
      onEntered={null}
      onEscKeyDown={null}
      onExited={null}
      onOverlayClick={null}
      open={false}
      overlayId={null}
      showCloseIcon={true}
      styles={Object {}}
    />
  </_default>
  <table
    className="table is-fullwidth is-tripped"
  >
    <thead>
      <tr
        className="notification"
      >
        <th
          colSpan={3}
        >
          Name
        </th>
        <th
          colSpan={2}
        >
          Proposal
        </th>
        <th
          colSpan={2}
        >
          Telescope
        </th>
        <th
          colSpan={3}
        >
          Start time
        </th>
      </tr>
    </thead>
    <tbody
      key="obsID-1"
    >
      <ObservationResults
        addAll={[Function]}
        cart={Array []}
        observation={
          Object {
            "files": Array [
              Object {
                "category": "cat-5",
                "dataType": "type-1",
                "declination": "20",
                "filename": "filename-1",
                "instrument": "RSS",
                "isReduced": true,
                "name": "name-1",
                "observationNight": "2010-01-01",
                "proposalCode": "code-1",
                "rightAscension": "20",
                "targetName": "target 001",
                "telescope": "SALT",
                "url": "./image0.jpg",
              },
            ],
            "id": "obsID-1",
            "name": "obs1",
            "proposal": "Code-1",
            "startTime": "2018-02-02 17:55:23",
            "telescope": "SALT",
          }
        }
        removeAll={[Function]}
      >
        <tr
          className="is-selected"
        >
          <td
            colSpan={3}
          >
            obs1
          </td>
          <td
            colSpan={2}
          >
            Code-1
          </td>
          <td
            colSpan={2}
          >
            SALT
          </td>
          <td>
            2018-02-02 17:55:23
          </td>
          <td>
            <button
              className="button is-info"
              name="addAll"
              onClick={[Function]}
            >
              add all
            </button>
          </td>
          <td
            colSpan={1}
          />
        </tr>
      </ObservationResults>
      <TableHead>
        <tr>
          <th>
            Add
          </th>
          <th>
            Filename
          </th>
          <th>
            Name
          </th>
          <th>
            Data type
          </th>
          <th>
            Raw/redused 
          </th>
          <th>
            Target name
          </th>
          <th>
            Right ascension
          </th>
          <th>
            Declination
          </th>
          <th>
            Category
          </th>
          <th>
            Instrument
          </th>
        </tr>
      </TableHead>
      <SearchRow
        addFile={[Function]}
        cart={Array []}
        files={
          Object {
            "category": "cat-5",
            "dataType": "type-1",
            "declination": "20",
            "filename": "filename-1",
            "instrument": "RSS",
            "isReduced": true,
            "name": "name-1",
            "observationNight": "2010-01-01",
            "proposalCode": "code-1",
            "rightAscension": "20",
            "targetName": "target 001",
            "telescope": "SALT",
            "url": "./image0.jpg",
          }
        }
        key="name-1"
        openModal={[Function]}
      >
        <tr
          key="name-1-row-123"
        >
          <td>
            <styled.input
              checked={false}
              onChange={[Function]}
            >
              <StyledComponent
                checked={false}
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [
                      Object {
                        "className": "checkbox",
                        "type": "checkbox",
                      },
                    ],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "c0",
                      "rules": Array [
                        "
  && {
    width: 18px;
    height: 18px;
  }
",
                      ],
                    },
                    "displayName": "styled.input",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "input",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                onChange={[Function]}
              >
                <input
                  checked={false}
                  className="checkbox c0"
                  onChange={[Function]}
                  type="checkbox"
                />
              </StyledComponent>
            </styled.input>
          </td>
          <td>
            <a
              onClick={[Function]}
            >
              filename-1
            </a>
          </td>
          <td>
            name-1
          </td>
          <td>
            type-1
          </td>
          <td>
            Redused
          </td>
          <td>
            target 001
          </td>
          <td>
            20
          </td>
          <td>
            20
          </td>
          <td>
            cat-5
          </td>
          <td>
            RSS
          </td>
        </tr>
      </SearchRow>
    </tbody>
    <tbody
      key="obsID-2"
    >
      <ObservationResults
        addAll={[Function]}
        cart={Array []}
        observation={
          Object {
            "files": Array [
              Object {
                "category": "cat-1",
                "dataType": "type-2",
                "declination": "20",
                "filename": "filename-2",
                "instrument": "RSS",
                "isReduced": true,
                "name": "name-2",
                "observationNight": "2010-01-01",
                "proposalCode": "code-2",
                "rightAscension": "20",
                "targetName": "target 002",
                "telescope": "SALT",
                "url": "./image4.jpg",
              },
            ],
            "id": "obsID-2",
            "name": "obs1",
            "proposal": "Code-2",
            "startTime": "2018-02-02 17:55:23",
            "telescope": "SALT",
          }
        }
        removeAll={[Function]}
      >
        <tr
          className="is-selected"
        >
          <td
            colSpan={3}
          >
            obs1
          </td>
          <td
            colSpan={2}
          >
            Code-2
          </td>
          <td
            colSpan={2}
          >
            SALT
          </td>
          <td>
            2018-02-02 17:55:23
          </td>
          <td>
            <button
              className="button is-info"
              name="addAll"
              onClick={[Function]}
            >
              add all
            </button>
          </td>
          <td
            colSpan={1}
          />
        </tr>
      </ObservationResults>
      <TableHead>
        <tr>
          <th>
            Add
          </th>
          <th>
            Filename
          </th>
          <th>
            Name
          </th>
          <th>
            Data type
          </th>
          <th>
            Raw/redused 
          </th>
          <th>
            Target name
          </th>
          <th>
            Right ascension
          </th>
          <th>
            Declination
          </th>
          <th>
            Category
          </th>
          <th>
            Instrument
          </th>
        </tr>
      </TableHead>
      <SearchRow
        addFile={[Function]}
        cart={Array []}
        files={
          Object {
            "category": "cat-1",
            "dataType": "type-2",
            "declination": "20",
            "filename": "filename-2",
            "instrument": "RSS",
            "isReduced": true,
            "name": "name-2",
            "observationNight": "2010-01-01",
            "proposalCode": "code-2",
            "rightAscension": "20",
            "targetName": "target 002",
            "telescope": "SALT",
            "url": "./image4.jpg",
          }
        }
        key="name-2"
        openModal={[Function]}
      >
        <tr
          key="name-2-row-123"
        >
          <td>
            <styled.input
              checked={false}
              onChange={[Function]}
            >
              <StyledComponent
                checked={false}
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [
                      Object {
                        "className": "checkbox",
                        "type": "checkbox",
                      },
                    ],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "c0",
                      "rules": Array [
                        "
  && {
    width: 18px;
    height: 18px;
  }
",
                      ],
                    },
                    "displayName": "styled.input",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "input",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                onChange={[Function]}
              >
                <input
                  checked={false}
                  className="checkbox c0"
                  onChange={[Function]}
                  type="checkbox"
                />
              </StyledComponent>
            </styled.input>
          </td>
          <td>
            <a
              onClick={[Function]}
            >
              filename-2
            </a>
          </td>
          <td>
            name-2
          </td>
          <td>
            type-2
          </td>
          <td>
            Redused
          </td>
          <td>
            target 002
          </td>
          <td>
            20
          </td>
          <td>
            20
          </td>
          <td>
            cat-1
          </td>
          <td>
            RSS
          </td>
        </tr>
      </SearchRow>
    </tbody>
  </table>
</SearchResults>
`;
